hail/python/hail/expr/aggregators/aggregators.py

```diff
@@ -896,8 +900,15 @@ def hardy_weinberg_test(expr) -> StructExpression:
     :class:`.StructExpression`
         Struct expression with fields `het_freq_hwe` and `p_value`.
     """
-    t = tstruct(het_freq_hwe=tfloat64, p_value=tfloat64)
-    return _agg_func('HardyWeinberg', [expr], t)
+    return hl.rbind(
+        hl.rbind(
+            expr,
+            lambda call: hl.agg.filter(call.ploidy == 2, hl.agg.counter(call.n_alt_alleles())
```

Comment:
it also occurs to me---I don't know if using `hl.agg._` is the right thing to do in any of these cases. We probably would accidentally end up with e.g.

```
hl.agg.filter(cond, hl.scan.count())
```

I think I've done this too, but it creates the wrong IR.

Reply:
hmm... Looking at this further, it does do the right thing, but it's definitely a bit odd...